homeassistant/auth_providers/homeassistant.py

```diff
@@ -118,16 +124,19 @@ class HassAuthProvider(auth.AuthProvider):
 
     DEFAULT_TITLE = 'Home Assistant Local'
 
-    async def async_credential_flow(self):
+    async def async_login_flow(self):
         """Return a flow to login."""
         return LoginFlow(self)
 
     async def async_validate_login(self, username, password):
         """Helper to validate a username and password."""
         data = Data(self.hass)
         await data.async_load()
-        await self.hass.async_add_executor_job(
-            data.validate_login, username, password)
+        try:
+            await self.hass.async_add_executor_job(
+                data.validate_login, username, password)
+        finally:
+            await data.async_save()
```

Comment:
Why do we need to always save?

Reply:
legacy code, should be removed. It was used in v1 implementation when auth module save data in  auth provider storage.